packages/synapse-react-client/src/utils/AppUtils/AppUtils.ts

```diff
@@ -1,11 +1,39 @@
 import { useHistory } from 'react-router-dom'
 import { LAST_PLACE_LOCALSTORAGE_KEY } from '../SynapseConstants'
 import { useEffect, useState } from 'react'
+import UniversalCookies from 'universal-cookie'
 
 export function storeLastPlace() {
   // save current route (so that we can go back here after SSO)
   localStorage.setItem(LAST_PLACE_LOCALSTORAGE_KEY, window.location.href)
 }
+const cookies = new UniversalCookies()
+export const ONE_SAGE_REDIRECT_COOKIE_KEY =
+  'org.sagebionetworks.cookies.redirect-after-login'
+
+export function storeRedirectURLForOneSageLogin() {
+  // save current URL in a cookie that One Sage will use to send you back to the correct page
+  const domainValue = window.location.hostname
+    .toLowerCase()
+    .endsWith('.synapse.org')
+    ? '.synapse.org'
+    : undefined
+  cookies.set(ONE_SAGE_REDIRECT_COOKIE_KEY, window.location.href, {
+    path: '/',
+    domain: domainValue,
+  })
+}
+
+export function processRedirectURLInOneSage() {
+  if (cookies.get(ONE_SAGE_REDIRECT_COOKIE_KEY)) {
+    const href = cookies.get(ONE_SAGE_REDIRECT_COOKIE_KEY)
+    cookies.remove(ONE_SAGE_REDIRECT_COOKIE_KEY)
+    window.location.assign(href)
+    return true
```

Comment:
This will never return

Reply:
That's what I thought, but I wasn't sure if this was an async process or not.  Back in the day, I saw a lot of instances of this line (which I think did have the possibility of executing the following lines):
window.location.href = "http://site.com"